packages/core/src/icon/icon.ts

```diff
@@ -1,36 +1,116 @@
 import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
+import type { IconPrefix, IconStyle } from '@fortawesome/fontawesome-common-types';
+import type { IconDefinition, IconName } from '@fortawesome/fontawesome-svg-core';
+import { findIconDefinition } from '@fortawesome/fontawesome-svg-core';
 import { LitElement, html } from 'lit';
 import { property } from 'lit/decorators.js';
 import { unsafeHTML } from 'lit/directives/unsafe-html.js';
 import styles from './icon.scss.js';
 
 export type IconResolver = (name: string) => string;
 
+export interface IconLibrary {
+  [key: string]: SLIconDefinition | CustomIconDefinition;
+}
+
+interface SLIconDefinition {
+  value?: string;
+  type?: string;
+  style?: string;
+  description?: string;
+}
+interface CustomIconDefinition extends SLIconDefinition {
+  svg: string;
+}
+
+declare global {
+  interface Window {
+    SLDS: {
+      icons: IconLibrary;
+    };
+  }
+}
+if (!window?.SLDS) {
+  window['SLDS'] = {
+    icons: {}
+  };
+}
+
 export class Icon extends LitElement {
   /** @private */
   static override styles: CSSResultGroup = styles;
 
-  /** @private */
-  static resolver: IconResolver = _ => 'No icon found';
+  static availableStyles: IconStyle[] = [];
+
+  /**
+   * Add icon(s) to the icon registry
+   *
+   * @param {IconDefinition | IconDefinition[] } faIcons One or more IconDefinition that have been imported from FontAwesome
+   */
+  static registerIcon(...faIcons: IconDefinition[]): void {
+    faIcons.forEach(icon => {
+      if (window.SLDS.icons[`${icon.prefix}-${icon.iconName}`]) {
+        console.warn(`Icon ${icon.prefix}-${icon.iconName} is already in the registry`);
```

Comment:
So what Lit does and Shoelace as well is that they have development/production environments. In development, the web components log various warnings etc to the console. In production they don't. It can be worth looking into how Shoelace did this and copy that :)